src/shoelace.ts

```diff
@@ -1,4 +1,30 @@
-// Cherry-picked Shoelace components:
-import "@shoelace-style/shoelace/dist/components/dropdown/dropdown.js";
-import "@shoelace-style/shoelace/dist/components/menu/menu.js";
-import "@shoelace-style/shoelace/dist/components/menu-item/menu-item.js";
+import kebabCase from "lodash/fp/kebabCase";
+
+// Cherry-picked Shoelace components
+// Import .component and register manually to avoid name collision
+// See https://github.com/shoelace-style/shoelace/issues/705
+import SlAlert from "@shoelace-style/shoelace/dist/components/alert/alert.component.js";
+import SlButton from "@shoelace-style/shoelace/dist/components/button/button.component.js";
+import SlCopyButton from "@shoelace-style/shoelace/dist/components/copy-button/copy-button.component.js";
+import SlDialog from "@shoelace-style/shoelace/dist/components/dialog/dialog.component.js";
+import SlDropdown from "@shoelace-style/shoelace/dist/components/dropdown/dropdown.component.js";
+import SlMenu from "@shoelace-style/shoelace/dist/components/menu/menu.component.js";
+import SlMenuItem from "@shoelace-style/shoelace/dist/components/menu-item/menu-item.component.js";
+
+// Shoelace components should only be registered if needed
+// to prevent issues with callers, e.g. Browsertrix.
+[
+  SlAlert,
+  SlButton,
+  SlCopyButton,
+  SlDialog,
+  SlDropdown,
+  SlMenu,
+  SlMenuItem,
+].forEach((slClass) => {
+  const name = kebabCase(slClass.name);
+
+  if (!customElements.get(name)) {
```

Comment:
According to: https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry/define#exceptions, it should throw if the element is already defined, and quickly testing locally seems to confirm this.

Reply:
Yeah this is probably unnecessary since Shoelace uses the same internal logic (https://github.com/shoelace-style/shoelace/blob/60927968503925f6887600b6dfb07f21b1735d9a/src/internal/shoelace-element.ts#L101) and https://github.com/webrecorder/browsertrix/pull/1861 is what seems to have solved the actual issue.

We could try the scoped customregistry polyfill instead: https://open-wc.org/docs/development/scoped-elements/